samza-core/src/main/java/org/apache/samza/storage/StorageManagerUtil.java

```diff
@@ -128,7 +128,11 @@ public static String readOffsetFile(File storagePartitionDir, String offsetFileN
 
     if (offsetFileRef.exists()) {
       LOG.info("Found offset file in storage partition directory: {}", storePath);
-      offset = FileUtil.readWithChecksum(offsetFileRef);
+      try {
+        offset = FileUtil.readWithChecksum(offsetFileRef);
```

Comment:
I think it would be better to add a unit test for these changes.

Reply:
Added the test for the case of offset already exist.